dynamiqs/core/diffrax_solver.py

```diff
@@ -34,10 +34,10 @@ def run(self) -> PyTree:
 
             if self.gradient is None:
                 adjoint = dx.RecursiveCheckpointAdjoint()
+            if isinstance(self.gradient, CheckpointAutograd):
+                adjoint = dx.RecursiveCheckpointAdjoint(self.gradient.ncheckpoints)
             elif isinstance(self.gradient, Autograd):
-                adjoint = dx.RecursiveCheckpointAdjoint()
-            elif isinstance(self.gradient, Adjoint):
-                adjoint = dx.BacksolveAdjoint()
+                adjoint = dx.DirectAdjoint()
```

Comment:
We should raise an error if the user provides with something else

Reply:
It's checked in sesolve/mesolve.